File: leetcode_nurtay/longest_palindrome.cpp
#include <bits/stdc++.h>
using namespace std;

bool Palindrome_Check(string s){
    for(int i = 0; i < s.size(); i++){
            if(s[i] != s[s.size()-i-1]) return false;
    }
    return true;
}

void solve(){
    string s; cin >> s;
    vector<string> palindromes;
    for(int i = 0; i < s.size(); i++){
        for(int j = 1; j <= s.size() - i; j++){
            string t = s.substr(i, j);
            if(Palindrome_Check(t)) palindromes.push_back(t);
        }
    }

    sort(palindromes.begin(), palindromes.end(), []
    (const std::string& first, const std::string& second){
        return first.size() < second.size();
    });

    cout << s.size();

    cout << palindromes[palindromes.size()-1];
}


int main(){
    solve();
}

File: leetcode_nurtay/longest_palindrome.py
s = input()
palindromes, i = [], 0
while i <= len(s):
    j = 0
    while j <= len(s):
        t = s[i:j]
        if t[::-1] == s[i:j] and len(t) != 0: palindromes.append(t)
        j += 1
    i += 1

palindromes.sort(key=len)
print(palindromes[-1])